### PR TITLE
Kelsha Darby - Fixes - User Story 45

### LINKED ISSUE #131: User Story 45
## Merchant Adds an Item

- The image text field is autofilled with the image placeholder
- The form accepts floats for both inventory and price (but later rounds them to 0)
    - So for price it needs to be greater than 0 and since .5 is it accepts it but later rounds it back to 0, breaking the rules required to create an item

### PR DESCRIPTION
**Pull request by kelshadarby**
## A Merchant User Can Add An Item
Closes #131 

**_Test_**
- Add to test to ensure that image field is not autofilled when a user opens the new item form
- Add new it block to test that a user cannot enter a float into inventory or price

**_Model_**
- Add to model to ensure that only integers are allowed

**_View_**
- Add conditional to image field to only fill in the field if the user has inputted text

### CODE REVIEW

File: app/models/item.rb [hunk 1]
```diff
@@ -10,8 +10,8 @@ class Item <ApplicationRecord
                         :price,
                         :inventory
   validates_inclusion_of :active?, :in => [true, false]
-  validates_numericality_of :price, greater_than: 0
-  validates_numericality_of :inventory, greater_than_or_equal_to: 0
+  validates_numericality_of :price, greater_than: 0, only_integer: true
+  validates_numericality_of :inventory, greater_than_or_equal_to: 0, only_integer: true
```
Comment (brian-greeson) [inline app/models/item.rb:14]: Nice!

File: app/views/merchant/items/new.html.erb [hunk 2]
```diff
@@ -9,7 +9,7 @@
 	<%= text_field_tag :description, "#{@item.description}" %>
 
 	<%= label_tag :image %>
-	<%= text_field_tag :image, "#{@item.image}" %>
+	<%= text_field_tag :image, ("#{@item.image}" if params[:image] != nil) %>
```
Comment (brian-greeson) [inline app/views/merchant/items/new.html.erb:12]: Interesting. I didn't know we had access to params in the view.

**Review by brian-greeson — APPROVED** [review]
Great Work!

File: app/models/item.rb (hunk 1 above)
Comment (fredrondina) [reply app/models/item.rb:14]: good catch

**Review by fredrondina — APPROVED** [review]
great work again

**Pull request merged**